com.microsoft.mrtk.input/Interactors/InteractorVisuals/BaseReticleVisual.cs

```diff
@@ -25,7 +25,21 @@ public class BaseReticleVisual : MonoBehaviour, IXRCustomReticleProvider
 
         public GameObject Reticle => customReticleAttached ? customReticle : baseReticle;
 
-        #region IXRCustomReticleProvider Implementation
+        private IVariableReticle variableReticle;
+
+        public IVariableReticle VariableReticle
```

Comment:
This feels a little bit like it doesn't belong in the base class, since it's a very specific type of reticle that not all visuals may implement.

Reply:
I ended up putting it in here because we need to update the variable reticle reference whenever a custom reticle is added or removed. It's not too much work to bring it out into the subclasses, but now they'll have to nominally override AddCustomReticle and RemoveCustomReticle.

I think we expect variable selectedness to be a pretty core part of XR interactions, so I'm a little hesitant to bring it out of the base class for organizational reasons. I can see actually making this a protected property instead of a public one, since I don't think we want other classes calling `UpdateVisuals` arbitrarily